src/ruby_supportlib/phusion_passenger/loader_shared_helpers.rb

```diff
@@ -210,6 +210,17 @@ def run_load_path_setup_code(options)
       #
       # The existence of Gemfile/gems.rb indicates whether (2) is true:
       elsif File.exist?('Gemfile') || File.exist?('gems.rb')
+        # Bundler 2.5.12+ throws a hissy fit if you don't have the
+        # version of bundler installed with which the Gemfile.lock was
+        # created. This causes extremely inscrutible error messages such
+        # as `/usr/bin/ruby doesn't exist`, so we cause RubyGems to raise
+        # an intelligible error here instead.
+        if File.exist?('Gemfile.lock')
+          lockfile_contents = File.read('Gemfile.lock').lines.map(&:strip)
+          index = lockfile_contents.index 'BUNDLED WITH'
+          bundler_version = lockfile_contents[index + 1] unless index.nil?
+          found = Gem::Specification.find_by_name('bundler', "= #{bundler_version}") unless bundler_version.nil?
```

Comment:
It appears you're not doing anything with `found`. Does the mere act of calling `Gem::Specification.find_by_name` fix this issue?